### DISCUSSION TITLE
Nessus scan questions

### DISCUSSION THREAD

**Discussion by steelwing** [Security-Onion-Solutions/securityonion#4388] [Unsupported Versions] [answered]
My Cyber/IA people ran scans of my new installation of SecurityOnion 2.3.50, and among the results that came back were these:
On port 443, Nessus found HTTP STS headers missing.  I can only imagine this refers to NGINX configuration, but I can't find where HSTS headers would be configured.
On port 5000, Nessus saw TLSv1 enabled and missing SSL certs.  We have certs in place for NGINX, but we don't know where to put them for whatever is on port 5000.  The only config file we've been able to identify as mentioning port 5000 is nginx.conf, and it only mentions TLSv1.2 and is pointing to the certs we've set up properly.

Any help you can offer would be greatly appreciated...

**Comment by mhgoff** [comment]
I am interested in this as well.

**Comment by launchdaemon** [comment]
I would imagine that the lack of a HSTS header is because by default the dashboard is configured with a self signed certificate. As a result if you're browser had seen the HSTS header it would terminate the connection if it could not trust the certificate.

**Reply by steelwing** [reply]
We've successfully replaced the certs for NGINX HTTPS, so that's no longer true.  Now how do we turn on the HSTS headers?

**Comment by defensivedepth** [comment, marked as answer]
https://github.com/Security-Onion-Solutions/securityonion/issues/4403 has been created to track setting the minimum TLS version for docker registry, which is on port 5000. To be clear, that port is only accessible to other nodes in the grid.

The HSTS header has already been implemented and will be included in the next release.

**Reply by steelwing** [reply]
So, to clarify:
* port 5000 is only exposed to other machines in the SO grid.  Nothing else sees that port.  (How did Nessus see the port in order get the results that it did, if this is the case?)
* TLS settings for that port will be fixed in v. 2.3.60 (per #4403 )
* HSTS is fixed in whatever release comes out next
Do I have that right?

**Reply by defensivedepth** [reply]
RE: TCP/5000 - yes, this is the case. Can you check and see how the scan was done? Was it an authenticated scan? Were there any other open ports?
RE: HSTS will be fixed in v. 2.3.60 as well

**Reply by steelwing** [reply]
5000 and 443 were the only ones reported with problems.  The scan was credentialed, so it's possible the port was only found because the scanner was effectively inside the OS.  I don't know precisely what the Nessus plugin that did that part of the scan does or looks for.
Thanks for clarifying the versions.  Do you have a timeframe for the release of 2.3.60?

**Reply by defensivedepth** [reply]
Yes, credentialed scan is how it found it for sure.

Here is an update on 2.3.60: https://blog.securityonion.net/2021/06/upcoming-security-onion-2360-release.html

**Comment by Shak3** [comment]
Hey @steelwing gonna go on a limb guess Mil? If so do you have RMF/ATO for SO? Trying to find/get one just for SO..we use under another ATO but that comes with other headaches..
